core/TempPath.php

```diff
@@ -11,8 +11,9 @@
 function getTempFolder($base = null) {
 	$parent = getTempParentFolder($base);
 
-	// The actual temp folder is a subfolder of getTempParentFolder(), named by username
-	$subfolder = $parent . DIRECTORY_SEPARATOR . getTempFolderUsername();
+	// The actual temp folder is a subfolder of getTempParentFolder(), named by username and suffixed with currently used php-version
+	$phpversion = '-php' . preg_replace('/[^\w-\.+]+/', '-', PHP_VERSION);
+	$subfolder = $parent . DIRECTORY_SEPARATOR . getTempFolderUsername() . $phpversion;
```

Comment:
I suggest to use DIRECTORY_SEPARATOR between silverstripe-cache and $phpversion. This is necessary so that we can have a parent (and predictable) folder name which can be controlled by a hard-coded `.htaccess` rewrite rule. If you make this variable then it's riskier to secure this folder.